app/services/post_bookmarkable.rb

```diff
@@ -0,0 +1,56 @@
+# frozen_string_literal: true
+
+class PostBookmarkable < BaseBookmarkable
+  MODEL = Post
+  SERIALIZER = UserPostBookmarkSerializer
+
+  def initialize
+    super(MODEL, SERIALIZER)
+    @preload_associations = [{ topic: [:topic_users, :tags] }, :user]
+  end
+
+  def list_query(user, guardian)
+    topics = Topic.listable_topics.secured(guardian)
+    pms = Topic.private_messages_for_user(user)
+    post_bookmarks = user
+      .bookmarks_of_type("Post")
+      .joins("INNER JOIN posts ON posts.id = bookmarks.bookmarkable_id AND bookmarks.bookmarkable_type = 'Post'")
+      .joins("LEFT JOIN topics ON topics.id = posts.topic_id")
+      .joins("LEFT JOIN topic_users ON topic_users.topic_id = topics.id")
+      .where("topic_users.user_id = ?", user.id)
+    guardian.filter_allowed_categories(
+      post_bookmarks.merge(topics.or(pms)).merge(Post.secured(guardian))
+    )
+  end
+
+  def search_query(bookmarks, query, ts_query, &bookmarkable_search)
+    bookmarkable_search.call(
+      bookmarks.joins(
+        "LEFT JOIN post_search_data ON post_search_data.post_id = bookmarks.bookmarkable_id AND bookmarks.bookmarkable_type = 'Post'"
+      ),
+      "#{ts_query} @@ post_search_data.search_data"
+    )
+  end
+
+  def reminder_handler(bookmark)
+    bookmark.user.notifications.create!(
+      notification_type: Notification.types[:bookmark_reminder],
+      topic_id: bookmark.bookmarkable.topic_id,
+      post_number: bookmark.bookmarkable.post_number,
+      data: {
+        title: bookmark.bookmarkable.topic.title,
+        display_username: bookmark.user.username,
+        bookmark_name: bookmark.name,
+        bookmarkable_url: bookmark.bookmarkable.url
+      }.to_json
+    )
+  end
+
+  def reminder_conditions(bookmark)
+    bookmark.bookmarkable.present? && bookmark.bookmarkable.topic.present?
```

Comment:
```suggestion
    bookmark.bookmarkable&.topic&.present?
```

Reply:
I changed this but then changed it back, because if `bookmark&.bookmarkable` is nil then we return `nil` here rather than `false` and I would rather return a boolean always. I could do `bookmark&.bookmarkable&.topic&.present? || false` but that looks messy.